### ISSUE TITLE
Save additional fields from articles 

### ISSUE THREAD

**Issue by mirestrepo** [JuliaHealth/BioMedQuery.jl#11]
* The PMCID and DOI (when they are listed)
* Journal information (including the ISSN, title, and abbreviation)
* The publication month

**Comment by mcmcgrath13** [comment]
Journal information and publication month will be included with the medline loader updates and could be incorporated into the efetch/search/save capabilities.

**Comment by mirestrepo** [comment]
I'm going to close this for now. We are still not getting 
* The PMCID and DOI (when they are listed)
But since it doesn't match the structure of general medline loader, we are not sure if we will implement it. If there is a need for it, we'll re-open

**Comment by msuperd1** [comment]
Thanks for updating! I notice that with the mesh_heading table the only mesh headings that are listed are those that have qualifiers. It would be helpful if mesh_headings without qualifiers were included as well.

**Comment by mcmcgrath13** [comment]
@msuperd1 mesh_headings without qualifiers should be included, but with default values of -1 for the qualifier fields.  Could you provide an example of where this isn't working as expected?

**Comment by msuperd1** [comment]
@mcmcgrath13 Thanks for the quick response. I should note that I've checked out master on BioMedQuery. I'm not sure if that changes anything. Here's what I'm running.

```
using BioServices.EUtils
using BioMedQuery.DBUtils
using EzXML
efetch_response = BioServices.EUtils.efetch(db = "pubmed", tool = "BioJulia", retmode = "xml", rettype = "null", id = [23889750])
xdoc = EzXML.parsexml(String(efetch_response.body))
xroot = root(xdoc)
password = ""
conn = BioMedQuery.DBUtils.init_mysql_database("127.0.0.1", "root", password, "test_db")
BioMedQuery.PubMed.create_tables!(conn)
BioMedQuery.PubMed.save_efetch!(conn, xroot)
```
In MySQL I get just 8 rows in my mesh_heading table, all of which have some qualifier.

Here is [pmid 23889750](https://www.ncbi.nlm.nih.gov/pubmed/?term=23889750) for reference.

**Comment by mirestrepo** [comment]
@msuperd1 There are significant changes to master as we now have DataFrames as the main (and intermediate) backend. I believe @mcmcgrath13 updated the basic documentations (follow links to latest instead of stable). Would you mind trying to use master? We'll tag a new version of this package once we're sure we have all the documentation updated.

**Comment by mirestrepo** [comment]
@mcmcgrath13 I think we should probably change the examples sooner rather than later. Should we start updating the notebooks before using Literate? or Do you think it's best to just do literate from beginning?

**Comment by mirestrepo** [comment]
@msuperd1 I tried running your code and even wrt the latest stable/tagged version it is outdated `save_efetch` used to take a dict....

```
using XMLDict
efetch_dict = parse_xml(String(efetch_response.body))
save_efetch!(conn, efetch_dict, verbose)
```
So I'm not sure what version of BioMedQuery you are running. In any case, instead of fixing outdated code, make sure to do 

```
Pkg.checkout("BioMedQuery")
Pkg.update()
```

If you haven't updated your Julia packages in a while, you may run into issues with MySQL. Let us know

**Comment by mcmcgrath13** [comment]
@msuperd1 I found the bug and am working on a fix - it was a clash with the MySQL foreign and primary key requirements.  I'll try to get master updated today.  In the mean time, if you remove the mesh_qual foreign key requirement from your MySQL mesh_heading table, it should work.  Also the dataframes option has all mesh headings available.

``` julia
using BioServices.EUtils
using BioMedQuery.DBUtils
using BioMedQuery.PubMed
using EzXML
efetch_response = BioServices.EUtils.efetch(db = "pubmed", tool = "BioJulia", retmode = "xml", rettype = "null", id = [23889750])
xdoc = EzXML.parsexml(String(efetch_response.data))
xroot = root(xdoc)
password = ""
conn = BioMedQuery.DBUtils.init_mysql_database("127.0.0.1", "root", password, "test_db")

## Option 1 to save to MySQL
BioMedQuery.PubMed.create_tables!(conn)
BioMedQuery.DBUtils.disable_foreign_checks(conn)
BioMedQuery.PubMed.save_efetch!(conn, xroot)

# Option 2 to get DataFrames only
dfs = BioMedQuery.PubMed.parse(xroot)
# (and maybe save to csv)
BioMedQuery.PubMed.dfs_to_csv(dfs, pwd(), "test_prefix_")
```

**Comment by mirestrepo** [comment]
@mcmcgrath13 I think the tests should maybe check the number of rows for the different fields for one article . ie 3 authors, 2mh .... that way if there are silent failures on the mysql side we can catch them

**Comment by mcmcgrath13** [comment]
@mirestrepo already added this for mesh_heading in the bug fix branch - I will expand it to loop through all tables

**Comment by msuperd1** [comment]
@mirestrepo Thank you! Yes I was using master, which is why I was feeding it save_efetch! an EzXML.Node. Sorry if there was confusion there. @mcmcgrath13 Thanks for your help. The BioMedQuery.DBUtils.disable_foreign_checks(conn) work around is working for me.

Separately, I'm getting errors with efetch_response.data: "ERROR: type Response has no field data." I'm not sure if this is a version issue with HTTP.jl. I've been able to use efetch_response.body just fine.

**Comment by mcmcgrath13** [comment]
@msuperd1 BioServices recently released a breaking change and BioMedQuery was using the prior version.  I'm also now fixing that.

**Comment by mcmcgrath13** [comment]
Waiting on a BioServices fix before these changes can be merged into master.

**Comment by mirestrepo** [comment]
@mcmcgrath13 BioServices has been merged and tagged so you should be able to merge this now

**Comment by mcmcgrath13** [comment]
This is merged into master. Closing the issue.
